x-pack/plugins/logstash/server/plugin.ts

```diff
@@ -56,7 +57,7 @@ export class LogstashPlugin implements Plugin {
   }
 
   start(core: CoreStart) {
-    const esClient = core.elasticsearch.legacy.createClient('logstash');
+    const esClient = core.elasticsearch.createClient('logstash');
```

Comment:
NIT: I wonder if it's enough with the _general_ client instead of the custom one.

Reply:
Only effect would be to change the log context for the ES operations of the plugin, but that was maybe here for that very reason. No sure who could answer that tbh.

Reply:
@ycombinator maybe? I'd remove the custom client. As an option, we can create an issue to remove it in `v8.0.0` if we consider the logger context name to be subject to semVer.

Reply:
cc @elastic/logstash could we get confirmation that it would be alright to just use the default ES client here? As said, the only difference is the logging context that is used to log potential ES-related entries (`elasticsearch.data.query` vs `elasticsearch.logstash.query`)

Reply:
I can't recall why the legacy client was being used here. In any case, I'm ++ to removing it if the only difference is the logging context.